spec/jobs/irs_attempts_events_batch_job_spec.rb

```diff
@@ -41,23 +41,31 @@
       end
 
       it 'batches and writes attempt events to an encrypted file' do
-        result = IrsAttemptsEventsBatchJob.perform_now(timestamp: start_time, dir_path: @dir_path)
-        expect(result[:file_path]).not_to be_nil
+        expect_any_instance_of(described_class).to receive(:create_and_upload_to_attempts_s3_resource)        
+        result = IrsAttemptsEventsBatchJob.perform_now(start_time)
 
-        file_data = File.open(result[:file_path], 'rb') do |file|
-          file.read
-        end
+        expect(result).not_to be_nil
+        expect(result[:filename]).not_to be_nil
+        expect(result[:iv]).not_to be_nil
+        expect(result[:encrypted_key]).not_to be_nil
+        expect(result[:requested_time]).to eq(start_time)
```

Comment:
I think you've got some good instincts about how to go about testing all this. It feels like we're in a really weird case in terms of what we're able to test here.

A few half-baked ideas for potential directions to take this:

We could stub out the call to EnvelopeEncryptor, and do something like:

```ruby
let(:encryptor_result) do
  IrsAttemptApi::EnvelopeEncryptor::Result.new(
    filename: 'foo',
    iv: 'sample_iv', etc...
  )
end

...

allow(IrsAttemptsApi::EnvelopeEncryptor).to receive(:encrypt)
  .with(data: 'some known value')
  .and_return(encryptor_result)
```

And then here we should be able to write something like:

```ruby
expect(result.filename).to eq(encryptor_result[:filename])
# etc.
```

I feel like this starts to be a lot, though. If that ends up feeling unwieldy, another option would be to just leave this, but we could make some of this more specific. We could, e.g., use the [match matcher](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/match-matcher):

```ruby
expect(result[:filename]).to match(/FCI-Logingov/)
```

I think it's OK for us to not use this test to assert that the EnvelopeEncryptor correctly encrypts things.